txpool/txpool.go

```diff
@@ -616,8 +626,32 @@ func (p *TxPool) validateTx(tx *types.Transaction) error {
 	}
 
 	// Reject underpriced transactions
-	if tx.IsUnderpriced(p.priceLimit) {
-		return ErrUnderpriced
+	if p.GetBaseFee() > 0 && tx.Type == types.DynamicFeeTx {
```

Comment:
Consider returning an error if a dynamic fee transaction is sent to the pool, but EIP-1559 hard fork is not enabled.

Reply:
@begmaroman take a look at this comment.

Reply:
Added check